shiny/module.py

```diff
@@ -24,25 +24,87 @@
 _: Id  # type: ignore
 
 
-@no_example()
+@add_example(ex_dir="api-examples/Module")
 def ui(fn: Callable[P, R]) -> Callable[Concatenate[str, P], R]:
+    """
+    Decorator for defining a Shiny module UI function.
+
+    This decorator allows you to write the UI portion of a Shiny module.
+    When your decorated `ui` function is called with an `id`,
+    the UI elements defined within will automatically be namespaced using that `id`.
+    This enables reuse of UI components and consistent input/output handling
+    when paired with a :func:`shiny.module.server` function.
+
+    Parameters
+    ----------
+    fn
+        A function that returns a Shiny UI element or layout (e.g., a `ui.panel_*` component).
+        This function should **not** accept an `id` parameter itself; the decorator injects it.
+
+    Returns
+    -------
+    :
+        A function that takes a `str` `id` as its first argument, followed by any additional
+        parameters accepted by `fn`. When called, it returns UI elements with input/output
+        IDs automatically namespaced using the provided module `id`.
+
+    See Also
+    --------
+    * Shiny Modules documentation: <https://shiny.posit.co/py/docs/modules.html>
+    * ~shiny.module.server
+    """
+
     def wrapper(id: Id, *args: P.args, **kwargs: P.kwargs) -> R:
         with namespace_context(id):
             return fn(*args, **kwargs)
 
     return wrapper
 
 
-@no_example()
+@add_example(ex_dir="api-examples/Module")
 def server(
     fn: Callable[Concatenate[Inputs, Outputs, Session, P], R],
 ) -> Callable[Concatenate[str, P], R]:
+    """
+    Decorator for defining a Shiny module server function.
+
+    This decorator is used to encapsulate the server logic for a Shiny module.
+    It automatically creates a namespaced child `Session` using the provided module `id`,
+    and passes the appropriate `input`, `output`, and `session` objects to your server function.
+
+    This ensures that the server logic is scoped correctly for each module instance and
+    allows for reuse of logic across multiple instances of the same module.
+
+    Parameters
+    ----------
+    fn
+        A server function that takes `input`, `output`, and `session` as its first
+        three arguments, followed by any additional arguments defined by the user.
+
+    Returns
+    -------
+    :
+        A function that takes a module `id` (as a string) as its first argument,
+        followed by any arguments expected by `fn`. When called, it will register
+        the module's server logic in a namespaced context.
+
+    See Also
+    --------
+    * Shiny Modules documentation: <https://shiny.posit.co/py/docs/modules.html>
+    * ~shiny.module.ui
```

Comment:
In this case I think it's helpful to see the full namespace in the link?

```suggestion
    * shiny.module.ui
```